sound/soc/sof/debug.c

```diff
@@ -17,6 +17,103 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+static ssize_t sof_dfsentry_write(struct file *file, const char __user *buffer,
+				  size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	struct snd_sof_dev *sdev = dfse->sdev;
+	struct sof_ipc_cmd_hdr hdr;
+	struct sof_ipc_reply reply;
+	size_t size;
+	char *string;
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG)
+	ktime_t start, end;
+	s64 ipc_response_time;
+	s64 avg_response_time = 0;
+	s64 min_response_time = LONG_MAX;
+	s64 max_response_time = 0;
+	int i, err, ret;
+	unsigned long ipc_count;
+#endif
+	string = kzalloc(count, GFP_KERNEL);
+	if (!string)
+		return -ENOMEM;
+
+	size = simple_write_to_buffer(string, count, ppos, buffer, count);
+	ret = size;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG)
+	/* check if "ipc_test" debugfs entry */
+	if (strcmp(dfse->dfsentry->d_name.name, "ipc_test"))
```

Comment:
ipc_test is actually used as an ipc_count, so should we change the entry to make it more self-explanatory, e.g. ipc_flood_count?